### ISSUE TITLE
no warning/error for unused string literals inside functions with "no-unused-expressions" rule

### ISSUE THREAD

**Issue by pzatrick** [eslint/eslint#4846]
First, thanks for creating such a useful tool and apologies if this issue has been raised before.

If I'm not mistaken, the following should produce an error or warning if the "no-unused-expressions" rule is enabled:

```javascript
function example () {
  "use strict";
  "hello world";
}

example();
```

Yet for me there is no output from eslint. However, the following does produce the correct warning on line 4:

```javascript
function example () {
  "use strict";
}
"hello world";

example();
```

This is my eslintrc.js:

```javascript
/* global module */
module.exports = {
  "rules": {
    "no-trailing-spaces": [ 1 ],
    "no-unused-expressions": [
      1,
      {
        allowShortCircuit: true,
        allowTernary: true
      }
    ],
    "vars-on-top": [ 1 ],
    "strict" : [
      2,
      "function"
    ],
    "indent": [
      2,
      2
    ],
    "quotes": [
      2,
      "double"
    ],
    "linebreak-style": [
      2,
      "unix"
    ],
    "semi": [
      2,
      "always"
    ]
  },
  "env": {
    "es6": true,
    "browser": true,
    "node": true
  },
  "extends": "eslint:recommended",
  "ecmaFeatures": {
    "jsx": true,
    "experimentalObjectRestSpread": true
  },
  "plugins": [
    "react"
  ]
};
```

My environment:
- Mac OS 10.10.5
- node v5.2.0
- eslint v1.10.3

**Comment by eslintbot** [comment]
@pzatrick Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
Does the presence or absence of `"use strict;"` make a difference? Is your first example flagged properly if the directive is removed?

**Comment by pzatrick** [comment]
With that change eslint only complains about the lack of "use strict":

```
/Path/to/example.js
  1:1  error  Use the function form of "use strict"  strict

✖ 1 problem (1 error, 0 warnings)
```

**Comment by pzatrick** [comment]
Also, it works as expected for other kinds of expressions:

```javascript
function example () {
  "use strict";
  5 + 5;
  "hi" + "world";
}

example();
```

eslint says:

```
/Path/to/example.js
  3:3  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  4:3  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions

✖ 2 problems (0 errors, 2 warnings)
```

**Comment by pzatrick** [comment]
And removing the "strict" rule has no effect on this apparent bug.

```javascript
function example () {
  "hello";
}

example();
```

eslint says: nothing

**Comment by pzatrick** [comment]
Ok, seems like eslint considers any unused string literal to be a possible directive like "use strict". I suppose that's okay since expressions that are just string literals don't occur very much outside of the strict mode directive.

**Comment by michaelficarra** [comment]
It's not just eslint, the language defines them as directives.
